File: web-framework/python/flask/src/route/debug.py
from logger.config import logger
import os

from flask import Blueprint, jsonify, request

debug_bp = Blueprint('debug', __name__, url_prefix='/debug')


@debug_bp.route('/displayHttpContext', methods=['GET', 'POST', 'PUT', 'DELETE'])
def echo():
    # 构建响应数据
    env = dict(os.environ)
    if 'ALIBABA_CLOUD_ACCESS_KEY_ID' in env:
        env['ALIBABA_CLOUD_ACCESS_KEY_ID'] = 'encrypted'
    if 'ALIBABA_CLOUD_ACCESS_KEY_SECRET' in env:
        env['ALIBABA_CLOUD_ACCESS_KEY_SECRET'] = 'encrypted'
    if 'ALIBABA_CLOUD_SECURITY_TOKEN' in env:
        env['ALIBABA_CLOUD_SECURITY_TOKEN'] = 'encrypted'

    result = {
        'path': request.path,
        'body': request.get_data(as_text=True),
        'method': request.method,
        'queries': request.args,
        'headers': dict(request.headers),
        'env': env,
    }

    # 记录日志信息
    logger.info('receive request: %s', result)

    # 返回 JSON 响应
    return jsonify(result)

File: web-framework/nodejs/express/src/myapp/routes/debug.js
const express = require('express');
const router = express.Router();
const {logger} = require('../logger/config')
/* GET home page. */
router.get('/displayHttpContext', (req, res) => {
    const env = Object.assign({}, process.env)
    env.ALIBABA_CLOUD_ACCESS_KEY_ID = "encrypted"
    env.ALIBABA_CLOUD_ACCESS_KEY_SECRET = "encrypted"
    env.ALIBABA_CLOUD_SECURITY_TOKEN = "encrypted"
    const headers = Object.assign({}, req.headers)
    headers['x-fc-access-key-id'] = "encrypted"
    headers['x-fc-access-key-secret'] = "encrypted"
    headers['x-fc-security-token'] = "encrypted"
    const result = {
        path: req.path,
        body: req.body,
        method: req.method,
        queries: req.query,
        headers: headers,
        env: env,
    };

    // 使用标准日志记录方式替换了console.log
    logger.info(`Receive request: ${JSON.stringify(result)}`);
    res.json(result);
});

module.exports = router;